datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -93,12 +93,17 @@ pub const UNNAMED_TABLE: &str = "?table?";
 /// ```
 #[derive(Debug, Clone)]
 pub struct LogicalPlanBuilder {
-    plan: LogicalPlan,
+    plan: Arc<LogicalPlan>,
 }
 
 impl LogicalPlanBuilder {
     /// Create a builder from an existing plan
     pub fn from(plan: LogicalPlan) -> Self {
+        Self { plan: Arc::new(plan) }
```

Comment:
A more "idomatic" way to do this would be call these methods 

```rust
impl LogicalPlanBuilder {
    pub fn new(plan: LogicalPlan) -> Self {...}
    pub fn new_from_arc(plan: Arc<LogicalPlan>) -> Self { ... }
...
}
```

And then implement `From` which would allow using `into()` syntax

```rust
impl From<LogicalPlan> for LogicalPlanBuilder { 
}
```

Reply:
Done